dht/dhtpeer/dhtpeer.go

```diff
@@ -1169,9 +1169,17 @@ func (dhtPeer *DHTPeer) RouteEnvelope(envel *aea.Envelope) error {
 			routeCount.Dec()
 
 			if !dhtPeer.isClosing {
-				lerror(err).Str("op", "route").Str("addr", target).
-					Msg("while rooting and looking up address on the DHT. moving it to she slow queue")
-				dhtPeer.slow_queue <- envel
+				if len(dhtPeer.slow_queue) >= SlowQueueSize-1 {
+					lerror(err).Str("op", "route").Str("addr", target).
+						Msg("slow queue is full: drop envelope")
+
+				} else {
+					lerror(err).Str("op", "route").Str("addr", target).
+						Msg("while rooting and looking up address on the DHT. moving it to she slow queue.")
+					dhtPeer.slow_queue <- envel
```

Comment:
What happens in the slow queue?
Do we eventually give up? 
Is there a time out?

Can the agent effectively reset their connection when they restart?

Reply:
@8ball030 
nope, there are a lot of design issues need to resolve later